core/tracker/tracker.go

```diff
@@ -146,9 +150,18 @@ func analyseDutyFailed(duty core.Duty, es []event) (bool, component, string) {
 	return true, events[0].component + 1, fmt.Sprintf("%s failed in %s component", duty.String(), (events[0].component + 1).String())
 }
 
-// failedDutyReporter instruments the duty. It ignores non-failed duties.
-// TODO(xenowits): Implement logic for reporting duties.
-func failedDutyReporter(core.Duty, bool, string, string) {}
+// failedDutyReporter instruments failed duties.
```

Comment:
logs and instruments failed duties?

Reply:
I would argue instrumentation is a umbrella term term that includes metrics and logs and tracing.